### PR TITLE
[Merged by Bors] - chore: remove duplicates about `List.append`

### PR DESCRIPTION
**Pull request by chabulhwi**
Replace duplicates in Mathlib with theorems in Lean core.

* `List.append_left_cancel`  → `List.append_cancel_left`
* `List.append_right_cancel` → `List.append_cancel_right`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/List/Basic.lean [hunk 1]
```diff
@@ -392,22 +392,18 @@ theorem cons_eq_append_iff {a b c : List α} {x : α} :
 
 #align list.append_inj_left' List.append_inj_left'ₓ -- implicits order
 
-theorem append_left_cancel {s t₁ t₂ : List α} (h : s ++ t₁ = s ++ t₂) : t₁ = t₂ :=
-  (append_right_inj _).1 h
-#align list.append_left_cancel List.append_left_cancel
+#align list.append_left_cancel List.append_cancel_left
```
Comment (jcommelin) [inline Mathlib/Data/List/Basic.lean:395]:
```suggestion
@[deprecated] alias append_left_cancel := append_cancel_left
#align list.append_left_cancel List.append_cancel_left
```

File: Mathlib/Data/List/Basic.lean [hunk 2]
```diff
@@ -392,22 +392,18 @@ theorem cons_eq_append_iff {a b c : List α} {x : α} :
...
 
-theorem append_right_cancel {s₁ s₂ t : List α} (h : s₁ ++ t = s₂ ++ t) : s₁ = s₂ :=
-  (append_left_inj _).1 h
-#align list.append_right_cancel List.append_right_cancel
+#align list.append_right_cancel List.append_cancel_right
```
Comment (jcommelin) [inline Mathlib/Data/List/Basic.lean:397]:
```suggestion
@[deprecated] alias append_right_cancel := append_cancel_right
#align list.append_right_cancel List.append_cancel_right
```

**Review by jcommelin — COMMENTED** [review]
Let's add deprecated aliases, in case these lemmas are used in downstream projects.

File: Mathlib/Data/List/Basic.lean [hunk 3]
```diff
@@ -392,22 +392,20 @@ theorem cons_eq_append_iff {a b c : List α} {x : α} :
 
 #align list.append_inj_left' List.append_inj_left'ₓ -- implicits order
 
-theorem append_left_cancel {s t₁ t₂ : List α} (h : s ++ t₁ = s ++ t₂) : t₁ = t₂ :=
-  (append_right_inj _).1 h
-#align list.append_left_cancel List.append_left_cancel
+@[deprecated] alias append_left_cancel := append_cancel_left
```
Comment (j-loreaux) [inline Mathlib/Data/List/Basic.lean:395]: please add a comment: `-- deprecated since 2024-01-18` to both aliases

**Review by j-loreaux — COMMENTED** [review]
Thanks! After fixing the issues below and waiting for CI to finish, you can merge.

bors d+

File: Mathlib/Data/List/Basic.lean (hunk 3 above)
Comment (chabulhwi) [reply Mathlib/Data/List/Basic.lean]: I added the comments to these aliases.

**Comment by chabulhwi** [comment]
bors d+

**Pull request closed without merging**